astropy/coordinates/builtin_frames/fk4_fk5_transforms.py

```diff
@@ -49,19 +52,19 @@ def fk4_no_e_to_fk5(fk4noecoord, fk5frame):
     pmat1 = fk4noecoord._precession_matrix(fk4noecoord.equinox, EQUINOX_B1950)
     pmat2 = fk5frame._precession_matrix(EQUINOX_J2000, fk5frame.equinox)
 
-    return pmat2 * B * pmat1
+    return matrix_product(pmat2, B, pmat1)
```

Comment:
Yes, I was quite happy with `matrix_product` too. I'll admit to being a bit less happy with `matrix_transpose`, but at least it isn't needed all that often...